### PR TITLE
fix the localStorage issue from #1915

### PR DESCRIPTION
**Pull request by fabianfrz**
Disables some features to keep the rest working if the browser has some features disabled.

### CODE REVIEW

**Comment by fichtner** [comment]
shouldn't we simply test for defined sessionStorage in place? I don't think the function helper gives us a lot of flexibility

**Comment by fichtner** [comment]
There is also more here... 

```
$ git grep sessionStorage
src/opnsense/www/js/opnsense.js:        sessionStorage.setItem('scrollpos', current_location()+"|"+$(window).scrollTop());
src/opnsense/www/js/opnsense.js:        var scrollpos = sessionStorage.getItem('scrollpos');
src/opnsense/www/js/opnsense_ui.js:            sessionStorage.setItem('all_help_preset', 1);
src/opnsense/www/js/opnsense_ui.js:            sessionStorage.setItem('all_help_preset', 0);
src/opnsense/www/js/opnsense_ui.js:    if (sessionStorage.getItem('all_help_preset') == 1) {
src/opnsense/www/js/opnsense_ui.js:    if (sessionStorage.getItem('show_advanced_preset') == 1) {
src/opnsense/www/js/opnsense_ui.js:            sessionStorage.setItem('show_advanced_preset', 1);
src/opnsense/www/js/opnsense_ui.js:            sessionStorage.setItem('show_advanced_preset', 0);
src/www/head.inc:          sessionStorage.setItem('all_help_preset', 1);
src/www/head.inc:          sessionStorage.setItem('all_help_preset', 0);
src/www/head.inc:      if (sessionStorage.getItem('all_help_preset') == 1) {
```

**Comment by fabianfrz** [comment]
@fichtner it is defined but it is `null`.

File: src/opnsense/www/js/opnsense.js [hunk 1]
```diff
@@ -36,9 +36,24 @@
   * @param value encoded text
   * @return string decoded text
   */
- function htmlDecode(value) {
-     return $("<textarea/>").html(value).text();
- }
+function htmlDecode(value) {
+    return $("<textarea/>").html(value).text();
+}
```
Comment (fabianfrz) [inline src/opnsense/www/js/opnsense.js]: @fichtner this had an one space indent

**Comment by fichtner** [comment]
I'm really against function to check by calling from multiple files, a portable null check is better as this will definitely break during weird caching or if we make a mistake later during refactor.

**Comment by fabianfrz** [comment]
@fichtner now without the function

**Comment by AdSchellevis** [comment]
@fabianfrz looks good to me too, thanks for fixing!

**Pull request merged**